### PR TITLE
Button color mappings

### LINKED ISSUE #11049: Rconsider the elevation -> token mapping objects/functions
Mapping objects like 
```ts
export const mapElevationToBackground: Record<Elevation, Color> = {
    '-1': 'backgroundSurfaceElevationNegative', // For example left menu is negative elevation
    0: 'backgroundSurfaceElevation0',
    1: 'backgroundSurfaceElevation1',
    2: 'backgroundSurfaceElevation2',
    3: 'backgroundSurfaceElevation3',
};
```

Are good, however it is a bit weird and not-intuitive to work with them. Maybe we should wrap them to some function or come up with some more elegant solution so it is more usable then stuff like this:

```ts
             return css`
                background: ${({ theme }) =>
                    theme[mapElevationToBackgroundTertiaryNormal[elevation]]};
                color: ${({ theme }) => theme.textOnTertiary};

                :hover,
                :active {
                    background: ${({ theme }) =>
                        theme[mapElevationToBackgroundTertiaryHover[elevation]]};
                }
            `;
```

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Description

- Fully type-safe and generic mapping for colors

Resolve https://github.com/trezor/trezor-suite/issues/11049

## Screenshots:

image
image

image

### CODE REVIEW

File: packages/components/src/components/buttons/buttonStyleUtils.ts [hunk 1]
```diff
@@ -11,48 +12,30 @@ export type ButtonVariant =
...
-
-export const mapElevationToBackgroundInfoHover: Record<Elevation, Color> = {
-    '-1': 'interactionBackgroundInfoDefaultHoverOnElevation3', // This is here just to satisfy Typescript mapping
+const mapElevationToButtonBackground = ({
+    elevation,
+    variant,
+    state,
+}: {
+    elevation: Elevation;
+    variant: 'destructive' | 'tertiary' | 'info' | 'warning';
```
Comment (dahaca) [inline packages/components/src/components/buttons/buttonStyleUtils.ts:23]: I would have the state type located elsewhere. And there already is the [`ButtonVariant`](https://github.com/trezor/trezor-suite/blob/8716ba727a73e972cd30680020dfea67f5e6a85d/packages/components/src/components/buttons/buttonStyleUtils.ts#L10) type, I would use it :)

File: packages/components/src/components/buttons/buttonStyleUtils.ts [hunk 2]
```diff
@@ -11,48 +12,30 @@ export type ButtonVariant =
...
+
+    const map: Record<Elevation, Color> = {
+        '-1': `interactionBackground${capitalizedVariant}DefaultHoverOnElevation3`, // For example left menu is negative elevation
+
+        // Button lies always on elevation so for example Button that lies has elevation 0, lies on elevation -1.
+        // This is why the values here a shifted by 1.
+        0: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevationNegative`,
+        1: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation0`,
+        2: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation1`,
+        3: `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation2`,
```
Comment (dahaca) [inline packages/components/src/components/buttons/buttonStyleUtils.ts:37]: How about `interactionBackground${capitalizedVariant}Default${capitalizedState}OnElevation${elevation}` here and in other places?

File: packages/components/src/components/buttons/buttonStyleUtils.ts [hunk 3]
```diff
@@ -132,24 +118,29 @@ export const getVariantStyle = (variant: ButtonVariant, elevation: Elevation) =>
...
+                        theme[
+                            mapElevationToButtonBackground({ elevation, variant, state: 'hover' })
+                        ]};
                 }
             `;
         case 'info':
             return css`
-                background: ${({ theme }) => theme[mapElevationToBackgroundInfoNormal[elevation]]};
+                background: ${({ theme }) =>
+                    theme[mapElevationToButtonBackground({ elevation, variant, state: 'normal' })]};
```
Comment (dahaca) [inline packages/components/src/components/buttons/buttonStyleUtils.ts]: I haven't fully checked but i believe if you change the function to 

```
({
    elevation,
    variant,
    state,
}) =>
({ theme }): CSSColor
```

You should be able to use it like:

```
background: ${mapElevationToButtonBackground( { elevation, variant, state: 'hover' } )}
````

Comment (dahaca) [reply packages/components/src/components/buttons/buttonStyleUtils.ts]: But it's sometimes used in a different way...

File: packages/components/src/components/Dropdown/Dropdown.tsx [hunk 4]
```diff
@@ -31,7 +31,6 @@ const Container = styled.div<{ disabled?: boolean }>`
     all: unset;
     width: fit-content;
     height: fit-content;
-    border-radius: 50%;
```
Comment (dahaca) [inline packages/components/src/components/Dropdown/Dropdown.tsx]: why is this removed?

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx [hunk 5]
```diff
@@ -49,6 +49,10 @@ const StyledButton = styled(Button)`
     margin: 4px 8px 4px 0;
 `;
 
+const Inline = styled.span`
+    display: inline-flex;
```
Comment (dahaca) [inline packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: why not simply style the `Button`? 👀

**Review by dahaca — COMMENTED** [review]
nice additions!

File: packages/components/src/components/buttons/buttonStyleUtils.ts (hunk 2 above)
Comment (peter-sanderson) [reply packages/components/src/components/buttons/buttonStyleUtils.ts:37]: Somethime it works like this in `mapElevationToBackground` however this has of 1 shift so its not that simple and I think its more readable like this.

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: it broke labels for example

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx (hunk 5 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: button needs to be `display: block` to make overflow: ellipsis work

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (dahaca) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: It was making the clickable area of the default toggle component match that component's shape:

image
image

So perhaps it should be present if the default toggle is there, at least? 🤔

File: packages/components/src/components/buttons/buttonStyleUtils.ts (hunk 3 above)
Comment (peter-sanderson) [reply packages/components/src/components/buttons/buttonStyleUtils.ts]: good point, I have implement it

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: Can you send me the place in the app where I can see this? I am not sure if I understand. Example would help. Thanks! :)

File: packages/components/src/components/buttons/buttonStyleUtils.ts (hunk 1 above)
Comment (peter-sanderson) [reply packages/components/src/components/buttons/buttonStyleUtils.ts:23]: done

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx (hunk 5 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: The filosofy here to make interface of component simple, if you need custom layouting of the inner components do it seperately

Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: Not the button itself but the `Content` within. Just an **nit** idea for a clearer style with less code :) 

```
> div:last-child {
    display: inline-flex;
}
```

**Review by jvaclavik — APPROVED** [review]
Nice 👏

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx (hunk 5 above)
Comment (peter-sanderson) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: Ah got it, instead of changing Button compont as a while you just suggest to use css magic instead of adding ann extra layer of components. 

I see, this is a tradeoff I am not sure about. @jvaclavik WDYT?

Comment (peter-sanderson) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: My opinion (not sure yetú is, that you shall not re-style the inner parts of the component from outside via CSS. This brakes the open-close principle.

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (dahaca) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: Any place where Dropdown uses the original toggle. And actually removing this introduces a bug, that was a reason why I put is there. Usually it's a good practice to ask a person who added the style when you see one you do not understand. 🙏 But even better to write the comment.

https://github.com/trezor/trezor-suite/assets/45338719/d571e02d-bf78-4ba9-bddd-808a3f8c0e1e

File: packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx (hunk 5 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Options/BitcoinOptions/BitcoinOptions.tsx]: It's all a nitpick ofc but I really like when JSX represents the actual visual structure, and each element has it's own clear place in DOM. 

Here, if you add another span, in DOM you will see `<button><span><span><span>name</span></span></span></button>` which is not ideal IMO :D `<span>` in Translation is also not ideal, actually, but this is a massive refactor. Ovb the ideal is to have `<button><span>name</span></button>` and have all styles consolidated in appropriate components. 

That massively improves readability and debugging and small things like this have potential to improve DX long term. I always try to simplify the tree structure because looking for styles and trying to undestand what's happening is instadeath 💀

File: packages/components/src/components/Dropdown/Dropdown.tsx (hunk 4 above)
Comment (peter-sanderson) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: Got it, thanks. Problem was it was just WTF, it was a hack with no comment to explain it.

This is what it does when you use custom children and not default `(...)`
image

Comment (peter-sanderson) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: I will prepare PR with proper fix.

Comment (peter-sanderson) [reply packages/components/src/components/Dropdown/Dropdown.tsx]: https://github.com/trezor/trezor-suite/pull/11107
